x-pack/libbeat/management/config.go

```diff
@@ -119,30 +47,34 @@ func defaultConfig() *Config {
 	}
 }
 
-type templateParams struct {
-	CentralManagementSettings string
-	BeatName                  string
+// Config returns a common.Config object holding the config from this block
+func (c *ConfigBlock) Config() (*common.Config, error) {
+	return common.NewConfigFrom(c.Raw)
 }
 
-// OverwriteConfigFile will overwrite beat settings file with the enrolled template.
-func (c *Config) OverwriteConfigFile(wr io.Writer, beatName string) error {
-	t := template.Must(template.New("beat.management.yml").Parse(ManagedConfigTemplate))
-
-	tmp := struct {
-		Management *Config `yaml:"management"`
-	}{
-		Management: c,
+// ConfigWithMeta returns a reload.ConfigWithMeta object holding the config from this block, meta will be nil
+func (c *ConfigBlock) ConfigWithMeta() (*reload.ConfigWithMeta, error) {
+	config, err := c.Config()
+	if err != nil {
+		return nil, err
 	}
+	return &reload.ConfigWithMeta{
+		Config: config,
+	}, nil
+}
 
-	data, err := yaml.Marshal(tmp)
+// ConfigBlocksEqual returns true if the given config blocks are equal, false if not
+func ConfigBlocksEqual(a, b ConfigBlocks) (bool, error) {
```

Comment:
I don't think that this is used by fleet, should it also be removed?

Reply:
If it's not being used, it should be removed. CI/compiler should complain if you removed too much (I hope).